### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR @mavericks-db  and I implemented the following:

- Created a Kanban board for task distribution
- Added Login page and registration page
- Added the Food list page to display a logged-in user's foods list
- Added the Recipe list page to display a logged-in user's recipes list
- Added a public recipes page to display a list of all public recipes ordered by the newest
- Added recipe details page to display details about a recipe
- Added General shopping list view
- Made sure there are no N+1 queries happening.
- Added navigation menu that allows users to open all of the pages.

### CODE REVIEW

File: app/controllers/recipe_foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  def create
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_food = @recipe.recipes_foods.create(quantity: params[:quantity], food_id: params[:food_id])
+    if @recipe_food.save
+      flash[:notice] = 'Food created successfully.'
+      redirect_to user_recipe_url(@recipe.user, @recipe)
+    else
+      render :new
+    end
+  end
```
Comment (dasileker) [inline app/controllers/recipe_foods_controller.rb:13]: - [x] Please make sure that the `@recipe_food` can add the food to the recipe, right now is generating an error of undefined.

Capture d’écran 2022-08-31 132350

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @EidHachem,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Linters are green ✅.
- Pr is very nice 👌.
- The readme file is nice ✅.
- Commit are consistent 👌.

## Required Changes ♻️
_Check the comments under the review._

- **Hello and nice job on your capstone but there is some minor issue encountered while reviewing your projects**one while adding ingredients it's returning undefined and the second is the test suits are failing*`*. so please fix those and re-submit again.**

Capture d’écran 2022-08-31 132826


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/recipe_foods_controller.rb (hunk 1 above)
Comment (mavericksbalitaan) [reply app/controllers/recipe_foods_controller.rb:13]: Done in [4efdd10](https://github.com/EidHachem/RecipeOnRails/pull/18/commits/4efdd10cd950b97b2629b47b443ce1a5c71aa522)

**Review by Davidosky007 — APPROVED** [review]
Hi @EidHachem @mavericks-db 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors ;
* All tests passed ✔️ 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**